overlays/nodejs/dynamodb/dynamodbMixins.ts

```diff
@@ -0,0 +1,106 @@
+
+import * as pulumi from "@pulumi/pulumi";
+import * as table from "./table";
+import * as iam from "../iam";
+import * as lambda from "../lambda";
+
+/**
+ * Arguments to control the event rule subscription.  Currently empty, but still defined in case of
+ * future need.
+ */
+export interface TableEventSubscriptionArgs {
+    /**
+     * The largest number of records that Lambda will retrieve from your event source at the time of invocation. Defaults to `100` for DynamoDB.
+     */
+    readonly batchSize?: number;
+
+    /**
+     * The position in the stream where AWS Lambda should start reading. Must be one of either `TRIM_HORIZON` or `LATEST`.
+     */
+    readonly startingPosition: "TRIM_HORIZON" | "LATEST";
+}
+
+export interface TableEvent {
+    Records: TableEventRecord[];
+}
+
+export interface TableEventRecord {
+    awsRegion: string;
+    dynamodb: {
+       ApproximateCreationDateTime: number;
+       Keys: Record<string, any>;
+       NewImage?: Record<string, any>;
+       OldImage?: Record<string, any>;
+       SequenceNumber: string;
+       SizeBytes: number;
+       StreamViewType: string;
+    };
+    eventID: string;
+    eventName: "INSERT" | "MODIFY" | "REMOVE";
+    eventSource: string;
+    eventVersion: string;
+    userIdentity: {
+       PrincipalId: string;
+       Type: string;
+    };
+}
+
+export type TableEventHandler = lambda.EventHandler<TableEvent, void>;
+
+export class TableEventSubscription extends lambda.EventSubscription {
+    public readonly table: pulumi.Output<table.Table>;
+    public readonly eventSourceMapping: lambda.EventSourceMapping;
+
+    constructor(
+        name: string, table: table.Table, handler: TableEventHandler,
+        args: TableEventSubscriptionArgs, opts?: pulumi.ResourceOptions) {
+
+        super("aws:dynamodb:TableEventSubscription", name, { table: table }, opts);
+
+        const parentOpts = { parent: this };
+        this.func = lambda.createFunctionFromEventHandler(name, handler, parentOpts);
+
+        this.permission = new lambda.Permission(name, {
+            function: this.func,
+            action: "lambda:InvokeFunction",
+            principal: "dynamodb.amazonaws.com",
+            sourceArn: table.streamArn,
+        }, parentOpts);
+
+        this.eventSourceMapping = new lambda.EventSourceMapping(name, {
+            batchSize: args.batchSize,
+            enabled: true,
+            eventSourceArn: table.streamArn,
+            functionName: this.func.name,
+            startingPosition: args.startingPosition,
+        }, parentOpts);
+    }
+}
+
+declare module "./table" {
+    interface Table {
+        /**
+         * Creates a new subscription to events fired from this Table to the handler provided, along
+         * with options to control the behavior of the subscription.
+         */
+        onEvent(name: string, handler: TableEventHandler,
+                args: TableEventSubscriptionArgs, opts?: pulumi.ResourceOptions): TableEventSubscription;
```

Comment:
bikeshedding. how does .onEvent sound to people here?   we could say .onTableChanged, but i don't think there's high value in changing hte name.